bliss/simulator/simulated_dataset.py

```diff
@@ -26,76 +24,72 @@
+        self.survey = survey
+        self.image_prior = self.survey.prior
+        self.background = self.survey.background
+        assert self.image_prior is not None, "Survey prior cannot be None."
+        assert self.background is not None, "Survey background cannot be None."
         self.image_prior.requires_grad_(False)
         self.background.requires_grad_(False)
+
+        assert survey.psf is not None, "Survey psf cannot be None."
+        self.image_decoder = ImageDecoder(
+            psf=survey.psf,
+        )
+
+        self.n_batches = n_batches
+        self.batch_size = self.image_prior.batch_size
         self.num_workers = num_workers
         self.fix_validation_set = fix_validation_set
         self.valid_n_batches = n_batches if valid_n_batches is None else valid_n_batches
 
         # list of (run, camcol, field) tuples from config
-        self.rcf_list = self._get_rcf_list(sdss_fields)
-
-    def _get_rcf_list(self, sdss_fields):
-        """Converts dict of row, camcol, field params into list of tuples."""
-        rcf_list = []  # list of (run, camcol, field) pairs
-        for rcf_params in sdss_fields["field_list"]:
-            run = rcf_params["run"]
-            camcol = rcf_params["camcol"]
-            rcf_list.extend([(run, camcol, field) for field in rcf_params["fields"]])
-        return np.array(rcf_list)
-
-    def get_random_rcf(self, num_samples=1):
+        self.image_ids = np.array(self.survey.image_ids())
+
+    def randomized_image_ids(self, num_samples=1):
         """Get random run, camcol, field combination from loaded params.
 
         Args:
             num_samples (int, optional): number of random samples to get. Defaults to 1.
 
         Returns:
-            Array of (row, camcol, field) pairs and index of each pair in self.rcf_list.
+            Array of (row, camcol, field) pairs and index of each pair in self.image_ids.
         """
-        n = np.random.randint(len(self.rcf_list), size=(num_samples,), dtype=int)
-        return self.rcf_list[n], n
+        n = np.random.randint(len(self.image_ids), size=(num_samples,), dtype=int)
+        return self.image_ids[n], n
 
-    @staticmethod
-    def _apply_noise(images_mean):
+    def _apply_noise(self, images_mean):
         # add noise to images.
         assert torch.all(images_mean > 1e-8)
         images = torch.sqrt(images_mean) * torch.randn_like(images_mean)
         images += images_mean
         return images
 
     def simulate_image(
-        self, tile_catalog: TileCatalog, rcf_indices
+        self, tile_catalog: TileCatalog, image_id_indices
     ) -> Tuple[Tensor, Tensor, Tensor, Tensor]:
         """Simulate a batch of images.
 
         Args:
             tile_catalog (TileCatalog): The TileCatalog to render.
-            rcf_indices: Indices of row/camcol/field in self.rcf_list to sample from.
+            image_id_indices: Indices of row/camcol/field in self.image_ids to sample from.
```

Comment:
This should probably just say "Indices in self.image_ids to sample from", without row/camcol/field

Reply:
Done.